contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,226 @@
+    Codelyzer lints your code against the [official Angular style guide](https://angular.io/guide/styleguide). With Angular CLI, you can simply run the command `ng lint` to get an analysis of whether your code follows the "best practices".
+        
+
+This [article](https://medium.freecodecamp.org/best-practices-for-a-clean-and-performant-angular-application-288e7b39eb6f) provides a comprehensive list of best tooling services offered by Angular.
+ 
+### Benefit 2: Angular Allows Cross-platform Development.
+
+Using the modern versions of Angular, developers can build applications that live on the web, mobile, or the desktop.
+
+* **Progressive Web Apps** - PWAs are web-based applications that contain most of the benefits of a typical mobile app. The [official documentation](https://developers.google.com/web/progressive-web-apps/) by Google describes PWAs to be reliable, fast and engaging. With Angular you can develop PWAs and provide an "immersive user experience".
+This [article](https://medium.com/@nsmirnova/creating-pwa-with-angular-5-e36ea2378b5d) is a good place to learn the basics of building PWAs using Angular.
+
+* **Native Apps** - Angular can be used to build <tooltip content="a smartphone application that is coded in a specific programming language, such as Swift for iOS or Java for Android operating systems">native mobile applications</tooltip>. There are many frameworks such as [NativeScript](https://www.nativescript.org/) and [Ionic](https://ionicframework.com/)
+that can be integrated with Angular for mobile application development. The [official Ionic website](https://ionicframework.com/docs/intro) claims that "integration with the Angular ecosystem is a breeze".
+
+* **Desktop Apps** - With Angular you can create desktop applications across Mac, Windows and Linux systems. In order for your application to be compatible with the different Operating Systems, you can use the <tooltip content="Electron is a framework that provides a set of APIs to interact with the Windows, OS X, and Linux operating systems. It makes cross-platform development easier.">[Electron](https://electronjs.org/)</tooltip> framework.
+You can integrate the Electron framework in your project using the Angular CLI and start building your application.
+This [article](https://medium.com/@yannmjl/how-to-build-native-cross-platform-desktop-apps-with-angular-electron-bd1d6e3919b2) is a good introduction to building desktop apps with Angular and Electron.
+
+### Benefit 3: Angular Provides Two-way Data Binding.
+
+Web apps have two main components, a _View_ and a _Model_. The View component is responsible for displaying data (i.e., the part users see). The Model is concerned with the logic implementation of the application. It updates and retrieves the data to and from the database.
```

Comment:
For "the part users see", since you use UI earlier, should you continue using UI instead of using this? Or you change the way to be more layman throughout the docs

Reply:
I think View is important to explain the concept of Data binding. It explains the concept more accurately I think